libraries/lib-channel/Channel.h

```diff
@@ -607,7 +607,7 @@ class CHANNEL_API ChannelGroup
 
 inline size_t Channel::NIntervals() const
 {
-   return GetChannelGroup().NIntervals();
+   return ReallyDoGetChannelGroup().NIntervals();
```

Comment:
What is it that I'm missing ...
Before this change:
```cpp
const ChannelGroup &Channel::GetChannelGroup() const
{
   assert(FindChannelIndex() >= 0);
   return DoGetChannelGroup();
}
```
After this change:
```cpp
ChannelGroup &Channel::ReallyDoGetChannelGroup() const
{
   return DoGetChannelGroup();
}
```
So what difference does it make?
I haven't been able to reproduce that bug. I suppose the clips don't actually need to be hours long? Mine were just a few seconds.<

Reply:
> What is it that I'm missing ...
`ReallyDoGetChannelGroup` is a virtual function 😎 and it's overwritten in `WaveChannel` :
```
ChannelGroup &WaveTrack::ReallyDoGetChannelGroup() const
{
   const Track *pTrack = this;
   if (const auto pOwner = GetHolder())
      pTrack = *pOwner->Find(this);
   const ChannelGroup &group = *pTrack;
   return const_cast<ChannelGroup&>(group);
}
```
GetHolder()->Find(this) may return track that is in fact different from this and that contains different amount of intervals.

> I haven't been able to reproduce that bug. I suppose the clips don't actually need to be hours long? Mine were just a few seconds.

For me crash occured on repaint event. So probably the length of the pasted clip is a important to reproduces this bug, because when you paste long clip UI thread isn't blocked and repaint events may slip into copying process.

Reply:
Ok, so there _is_ a need for the track to be stereo for the crash to happen, right? Paste happens, but the process is slow and repainting happens before the left channel is finished copied. At this stage, left channel has size 1, and right channel, 0, but the painting thinks both have size 1. Could that be the problem? I came across that problem in the recent past.

Reply:
Actually what I said doesn't make sense. If asking `NIntervals()` on the right channel of that yet incomplete stereo track, the painting would have got 0, which would be the truth ... Now it would say `1`, but that wouldn't be the truth anymore. So I wonder if fixing this place wouldn't break another one.
Do you know exactly which line crashed? I'll try to reproduce.

Reply:
Caught it: 
![image](https://github.com/audacity/audacity/assets/22740106/5f49d960-b0ce-436d-8ffb-6d4e9f7a59a0)

Reply:
Hm, I think you are right and the hotfix you are suggesting is correct and mine is not. I didn't considered that the problem comes from invariant violation. Obviously track could be inspected while being modified.

Reply:
I see Matt's long comment.  I expect this crash is not reproducible in the wave-clip-refactoring branch because of changes from channel-major to interval-major in pasting.

If this hot fix is merged to master, and master is then merged to wave-clip-refactoring to keep it current, the fix should then be reverted in that branch.